aten/src/ATen/native/ForeachUtils.h

```diff
@@ -54,5 +54,48 @@ bool check_fast_route(TensorList tensors, Scalar scalar) {
   return true;
 }
 
-} // namespace
+void verify_list(TensorList tensors1, TensorList tensors2) {
+  TORCH_CHECK(tensors1.size() > 0, "Tensor list must have at least one tensor.");
+  TORCH_CHECK(tensors2.size() > 0, "Tensor list must have at least one tensor.");
+  TORCH_CHECK(tensors1.size() == tensors2.size(), "Tensor lists must have the same number of tensors.");
+
+  auto expected_dtype = tensors1[0].dtype();
+
+  for (int i = 0; i < tensors1.size(); i++) {
+    TORCH_CHECK(tensors1[i].dtype() == expected_dtype, "All tensors in the tensor list must have the same dtype.");
+    TORCH_CHECK(tensors2[i].dtype() == expected_dtype, "All tensors in the tensor list must have the same dtype.");
+    TORCH_CHECK(tensors1[i].sizes() == tensors2[i].sizes(), "Corresponding tensors in lists must have the same size.");
```

Comment:
nit: It's kind of nice to report to the user what quantities existed in their code so that they're easier to debug. e.g. "Tensor lists must have the same number of tensors, got 3 and 5"